Fails with:
```
[231/262] Building C object src/nvim/CMakeFiles/nvim.dir/profile.c.o
FAILED: /usr/bin/gcc-4.8 -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -D_GNU_SOURCE -Iconfig -I../src -Isrc/nvim/auto -Iinclude -I/home/travis/nvim-deps/usr/include/luajit-2.0 -isystem /home/travis/nvim-deps/usr/include -g   -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector --param ssp-buffer-size=4 -Werror -MD -MT src/nvim/CMakeFiles/nvim.dir/profile.c.o -MF src/nvim/CMakeFiles/nvim.dir/profile.c.o.d -o src/nvim/CMakeFiles/nvim.dir/profile.c.o   -c ../src/nvim/profile.c
../src/nvim/profile.c: In function ‘profile_setlimit’:
../src/nvim/profile.c:70:3: error: conversion to ‘long unsigned int’ from ‘proftime_T’ may change the sign of the result [-Werror=sign-conversion]
   STRICT_ADD(os_hrtime(), nsec, &rv, int64_t);
   ^
cc1: all warnings being treated as errors
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -323,18 +323,6 @@ if(HAS_DIAG_COLOR_FLAG)
   endif()
 endif()
 
-if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
-  # 1. Array-bounds testing is broken in some GCC versions before 4.8.5.
-  #    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56273
-  # 2. But _Pragma("...ignored") is broken (unresolved) in GCC 5+:
-  #    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66099
-  # So we must disable -Warray-bounds globally for GCC (for kbtree.h, #7083).
-  check_c_compiler_flag(-Warray-bounds HAS_WARRAY_BOUNDS)
-  if(HAS_WARRAY_BOUNDS)
-    add_compile_options(-Wno-array-bounds)
-  endif()
-endif()
-
 option(TRAVIS_CI_BUILD "Travis/QuickBuild CI, extra flags will be set" OFF)
 
 if(TRAVIS_CI_BUILD)

